src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -186,6 +191,8 @@ private JoinableTask ExecuteWithinLockAsync(Func<Task> task)
             AggregateWorkspaceProjectContext previousContextToDispose = null;
             return await ExecuteWithinLockAsync(async () =>
             {
+                await _commonServices.ThreadingService.SwitchToUIThread();
```

Comment:
What's the rationale for this part of the change, especially since you `ConfigureAwait(false)` two lines down?